lib/ansible/plugins/inventory/onesixtyone.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) 2019, Victor da Costa <vdacosta@redhat.com>
+# Copyright (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: onesixtyone
+    plugin_type: inventory
+    version_added: "2.9"
+    short_description: Uses onesityone to find hosts to target
+    description:
+        - Uses a YAML configuration file with a valid YAML extension.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    requirements:
+      - onesixtyone CLI installed
+      - python-gevent installed
+    options:
+        addresses:
+            description: List of CIDRs and IPs to scan.
+            type: list
+            required: false
+            default:
+              - 192.168.0.0/24
+        communities:
+            description: List of communities to query per device.
+            type: list
+            required: false
+            default:
+              - "public"
+        wait:
+          description: wait n milliseconds (1/1000 of a second) between sending packets.
+          type: int
+          default: 10
+          required: false
+        platform:
+          description: Dict containing additional mappings between Platform (key) and regex to match in sysDescr (value).
+          type: dict
+          default: {}
+          required: false
+    notes:
+      - Only discover devices running SNMPv2.
+      - TODO: Add cache support.
+'''
+EXAMPLES = '''
+    # onesixtyone.yaml or any other YAML file-extension
+    plugin: onesixtyone
+    strict: False
+    addresses:
+      - 192.168.0.0/24
+'''
+
+
+import os
+import re
+import tempfile
+import subprocess
+
+from ansible import constants as C
+from ansible.errors import AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+try:
+  import gevent
+  from gevent import socket
+except:
+  raise Exception('this inventory plugin requires the gevent library. Try: yum install python-gevent')
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'onesixtyone'
+
+    def __init__(self):
+
+        self._onesixtyone = None
+        for path in os.environ.get('PATH').split(':'):
+            candidate = os.path.join(path, 'onesixtyone')
+            if os.path.exists(candidate):
+                self._onesixtyone = candidate
+                break
+
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+
+        if super(InventoryModule, self).verify_file(path):
```

Comment:
Use "guard expression" style here to reduce nesting
```suggestion
        if not super(InventoryModule, self).verify_file(path):
            return False
```

Reply:
```PYTHON
    def verify_file(self, path):

        if not super(InventoryModule, self).verify_file(path):
            return False

        for suffix in ('yaml', 'yml'):
            maybe = '{fn}.{suffix}'.format(fn=self.NAME, suffix=suffix)
            if path.endswith(maybe):
                return True
```